conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/GramineDirectEnclaveEnvironment.kt

```diff
@@ -28,103 +30,86 @@ class GramineEnclaveEnvironment(
         require(enclaveMode != EnclaveMode.MOCK) { "Gramine can't run in MOCK mode" }
     }
 
-    private val tcbLevel = 1
-
-    private val currentCpuSvn: ByteArray by lazy {
-        versionToCpuSvn(tcbLevel)
-    }
-
-    /** Generate simulated mrenclave value by hashing the enclave fat-jar */
-    private val mrenclave: ByteArray by lazy {
-        val digest = MessageDigest.getInstance("SHA-256")
-
-        val buffer = ByteArray(65536)
-        var bytesRead: Int
-
-        enclaveClass.protectionDomain.codeSource.location.openStream().use {
-            bytesRead = it.read(buffer)
-            while (bytesRead >= 0) {
-                digest.update(buffer, 0, bytesRead)
-                bytesRead = it.read(buffer)
-            }
-        }
-
-        digest.digest()
-    }
-
-    private val aesSealingKey by lazy(LazyThreadSafetyMode.NONE) {
-        digest("SHA-256") { update(enclaveClass.name.toByteArray()) }.copyOf(16)
-    }
-
     override fun createReport(
-            targetInfo: ByteCursor<SgxTargetInfo>?,
-            reportData: ByteCursor<SgxReportData>?
+        targetInfo: ByteCursor<SgxTargetInfo>?,
+        reportData: ByteCursor<SgxReportData>?
     ): ByteCursor<SgxReport> {
+        check(simulationMrsigner != null) { "Simulation MRSigner not provided in the enclave" }
```

Comment:
There's no reason now for simulationMrsigner to be nullable.

Likewise, this class will always represent simulation mode, there's no need to pass in the mode into the c'tor.

Reply:
Correct, this is a leftover of the copy of the class, removing it

Reply:
Removed that check

Reply:
But I added one in the class where this is called, to verify that the signer has been retrieved correctly, let me know if this is fine